### PR TITLE
Raise semantic exception when table does not have columns

### PR DESCRIPTION
**Pull request by kokosing**
Raise semantic exception when table does not have columns

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java [hunk 1]
```diff
@@ -744,6 +744,10 @@ protected Scope visitTable(Table table, Optional<Scope> scope)
                                 .collect(toImmutableList());
                     }
 
+                    if (fields.isEmpty()) {
+                        throw new SemanticException(MISSING_COLUMN, table, "Table '%s' does not have any columns", table.getName());
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java]: This is actually a connector error and not a semantic error for the user's query.

I think we should add a new `INVALID_TABLE` user error (analogous to `INVALID_VIEW`). That check should go in `MetadataManager.getTableMetadata()`, so it covers all usages. We turn the check here into a `verify()` (if it happens, there is a bug elsewhere).

**Comment by kokosing** [comment]
@electrum comments applied

**Comment by kokosing** [comment]
Rebased.

@electrum Ping.

@findepi Would you like to take a look?

File: presto-main/src/main/java/com/facebook/presto/metadata/MetadataManager.java [hunk 2]
```diff
@@ -377,6 +378,10 @@ public TableMetadata getTableMetadata(Session session, TableHandle tableHandle)
             throw new PrestoException(NOT_SUPPORTED, "Table has no columns: " + tableHandle);
         }
 
+        if (tableMetadata.getColumns().isEmpty()) {
+            throw new PrestoException(INVALID_TABLE_SCHEMA, format("Table '%s' should contain at least single column", tableHandle));
```
Comment (findepi) [inline presto-main/src/main/java/com/facebook/presto/metadata/MetadataManager.java]: There already is `if (tableMetadata.getColumns().isEmpty()) {` check above 😉

File: presto-main/src/main/java/com/facebook/presto/metadata/MetadataManager.java [hunk 3]
```diff
@@ -395,6 +400,10 @@ public TableStatistics getTableStatistics(Session session, TableHandle tableHand
         ConnectorMetadata metadata = getMetadata(session, connectorId);
         Map<String, ColumnHandle> handles = metadata.getColumnHandles(session.toConnectorSession(connectorId), tableHandle.getConnectorHandle());
 
+        if (handles.isEmpty()) {
+            throw new PrestoException(INVALID_TABLE_SCHEMA, format("Table '%s' should contain at least single column", tableHandle));
```
Comment (findepi) [inline presto-main/src/main/java/com/facebook/presto/metadata/MetadataManager.java]: make exception the same as in `getTableMetadata()`

**Pull request closed without merging**